plugins/modules/dcnm_res_manager.py

```diff
@@ -0,0 +1,1254 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2020-2022 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+__author__ = "Mallik Mudigonda"
+
+DOCUMENTATION = """
+---
+module: dcnm_res_manager
+short_description: DCNM ansible module for managing resources.
+version_added: "2.1.0"
+description:
+    - DCNM ansible module for creating, deleting and querying resources
+author: Mallik Mudigonda (@mmudigon)
+options:
+  fabric:
+    description:
+      - 'Name of the target fabric for resource manager operations'
+    type: str
+    required: true
+  state:
+    description:
+      - The required state of the configuration after module completion.
+    type: str
+    required: false
+    choices:
+      - merged
+      - deleted
+      - query
+    default: merged
+  config:
+    description:
+      - A list of dictionaries containing resources and switch information
+    type: list
+    elements: dict
+    suboptions:
+      entity_name:
+        description:
+          - A unique name which identifies the entity to which the resourcce is allocated to.
+          - The format of this parameter depends on the scope_type. The details are provided in
+          - the EXAMPLES section
+        type: str
+        required: true
+      pool_type:
+        description:
+          - Type of resource pool
+        type: str
+        required: true
+        choices:
+          - ID
+          - IP
+          - SUBNET
+      pool_name:
+        description:
+          - Name of the resource pool from which the resource is allocated
+        type: str
+        required: true
+      scope_type:
+        description:
+          - Socpe of resource allocation
+        type: str
+        required: true
+        choices:
```

Comment:
Maybe I missed this in the code but are we enforcing these choices in the module?

Reply:
Yes, We have a routine which checks for allowed combinations